### PR TITLE
Make syntax of generic overload variants in messages close to PEP 695

### PR DESCRIPTION
**Pull request by JukkaL**
We used a custom syntax for type variable bounds and restrictions. Use PEP 695 syntax instead (or at least something closer to PEP 695 syntax).

### CODE REVIEW

**Comment by JukkaL** [comment]
This conflicts with #17392. I will wait until #17392 is merged and will fix the conflicts here.

**Review by ilevkivskyi — APPROVED** [review]

**Comment by davidhalter** [comment]
@JukkaL Just FYI: I think it's a bit confusing that `TypeStrVisitor.visit_callable_type` still uses the old syntax (this causes old syntax in all `reveal_type()` calls). I have unfortunately no time to fix it now.

**Pull request merged**
